### ISSUE TITLE
JIT performance discrepancies

### ISSUE THREAD

**Issue by alextrekov3307** [pytorch/pytorch#32754]
PyTorch Nightly (today)
Fedora 31
Anaconda Python 3.7
GPU: 970 (staying under 3.5 GB limit, last 0.5GB is lower quality memory)

Not completely confident it's a bug, if it isn't there should certainly be something mentioned in the documentation.

`import numpy as np
import torch
import time

data_type = torch.torch.cuda.DoubleTensor

cuda = torch.device('cuda')

points = torch.rand((2**22,3), dtype=torch.double, device=cuda, requires_grad=False).type(data_type)
points[:, 2] = 0.0

@torch.jit.script
def attractor_func(v):
	x, y, z = torch.chunk(v, 3, dim=1)

	#lorenz
	a = 10.0
	b = 28.0
	c = 8.0 / 3.0
	dx = a * (y - x)
	dy = x * (b - z) - y
	dz = x * y - c * z

	return torch.cat([dx, dy, dz], 1)

@torch.jit.script
def rk4(v):
	h = 0.001
	k1 = attractor_func(v)
	k2 = attractor_func(v + (h / 2.0) * k1)
	k3 = attractor_func(v + (h / 2.0) * k2)
	k4 = attractor_func(v + h * k3)
	return h * (k1 / 6.0 + k2 / 3.0 + k3 / 3.0 + k4 / 6.0)

rk4 = torch.jit.trace(rk4, example_inputs=points)

start = time.time()
for _ in range(1024):
	points += rk4(points)
end = time.time()
print(end - start)`

output:
18.279905796051025

comment out the trace, output:
18.649290084838867

Why is the trace adding performance to JIT scripts?

Uncomment the trace and comment the script tags on both functions, output:
28.479955911636353

Why does the trace benefit greatly from JIT scripts?

Another example that may have more to do with wrong function choice or some minor inefficiency somewhere:

`import numpy as np
import torch
import time

data_type = torch.torch.cuda.DoubleTensor

cuda = torch.device('cuda')

points = torch.rand((2**22,3), dtype=torch.double, device=cuda, requires_grad=False).type(data_type)
points[:, 2] = 0.0

def attractor_func(v):
	x, y, z = torch.chunk(v, 3, dim=1)

	#duffing
	a = 0.25
	b = 0.3
	c = 1.0
	dx = y
	dy = x - x**3 - a * y + b * torch.cos(c * z)
	dz = x - x + 1.0
	return torch.cat([dx, dy, dz], 1)

def rk4(v):
	h = 0.001
	k1 = attractor_func(v)
	k2 = attractor_func(v + (h / 2.0) * k1)
	k3 = attractor_func(v + (h / 2.0) * k2)
	k4 = attractor_func(v + h * k3)
	return h * (k1 / 6.0 + k2 / 3.0 + k3 / 3.0 + k4 / 6.0)

rk4 = torch.jit.trace(rk4, example_inputs=points)

start = time.time()
for _ in range(1024):
	points += rk4(points)
end = time.time()
print(end - start)`

output:
46.12030529975891

change to `dz = x.clone().fill_(1.0)`

output:
65.20094799995422

change to `dz = torch.ones((2**22,1), dtype=torch.double, device=cuda, requires_grad=False).type(data_type)`

output:
46.92164850234985

change to `dz = x.new_ones((2**22,1))`

output:
46.9119393825531

Only the last two make sense. Thought it seems like it’s precisely the last two that make the use of a script tag for the function impossible. In this case, unlike the previous one the script tags don't appear to do much.

**Comment by suo** [comment]
That's indeed pretty weird, thanks for the clear repro. I'll try to take a look at this when I get a sec

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
